### ISSUE TITLE
Feature request: cli support for filtering by rule metadata

### ISSUE THREAD

**Issue by Vasco-jofra** [semgrep/semgrep#6752]
**Is your feature request related to a problem? Please describe.**
I want to have a single semgrep-rules repository with high-precision rules (`subcategory: vuln`) and low-precision rules that require human eyes to determine their validity (`subcategory: audit`). I would like users to be able to easily use just the high-precision rules or all of them in different use cases (e.g., CI vs doing security assessments).

**Describe the solution you'd like**
I would like the Semgrep cli to filter rule metadata (currently, you can filter only by `--severity` against `INFO`, `WARNING`, and `ERROR`). This feature would enable running a command such as: `semgrep --subcategory vuln --impact HIGH --confidence HIGH`. I think most default metadata fields (as defined in https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/#including-fields-required-by-security-category) should be filterable in the cli:
 - `technology`
 - `category`
 - sub-metadata of the `security` category:
     - `confidence`
     - `likelihood`
     - `impact`
     - `subcategory`

**Comment by aryx** [comment]
We're planning to allow people to write rules using jsonnet. This would solve your problem.
See for example how we use jsonnet internally to specify rules to run: https://github.com/returntocorp/semgrep/blob/develop/semgrep.jsonnet

**Comment by Vasco-jofra** [comment]
That would definitely help! Especially when running it in a CI. I would still like to be able to filter directly in the cli without writing the jsonnet file, but it's good to know that there's a way around it now.

**Comment by aryx** [comment]
We would probably not add flags in the CLI. There are too many possible way to filter rules to bother to add as many flags. That's the beauty of jsonnet, it's programmable, so you can filter however you want, by impact, confidence, date, ranking, whatever.

**Comment by Vasco-jofra** [comment]
Yeah, not adding all those flags in CI is definitely a good call. I would implement it as `--filter impact HIGH` to avoid adding too many flags. Alternatively, you could maybe do `--jsonnet "{'impact': 'HIGH'}"` (not sure what the syntax is)

IMO, the downside of jsonnet is that it's another thing you need to learn, and you can't just easily use it in the cli. For example, I might run some rules in the cli, get many results, and want to filter by HIGH impact to see the most important ones. For this, I would have to write a whole jsonnet file. I know that this is not a common use case, so feel free to close this issue if it is not something that you plan to support.

**Comment by aryx** [comment]
In theory you could use semgrep itself to specify which rules to use, since you can write yaml rules.

**Comment by disconnect3d** [comment]
Is there any update to this?
